### PR TITLE
Feature/331: Add new scoring component

### PR DESCRIPTION
**Pull request by lkleisa**

### CODE REVIEW

**Comment by lkleisa** [comment]
Ticket für die Tests als E2E zu machen: https://github.com/puzzle/okr/issues/416

File: frontend/src/app/confidence/confidence.component.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+p {
+  font-size: 12px;
```
Comment (peggimann) [inline frontend/src/app/confidence/confidence.component.scss:2]: change to rem

File: frontend/src/app/shared/scoring/scoring/scoring.component.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        [style.--data-bar-color]="failColor"
+        [attr.data-end-line]="endLineFail"
+      ></div>
+    </div>
+
+    <div class="d-flex flex-column">
+      <p class="text-center sub-title">Commit</p>
+      <div
+        id="{{ 'commit-card' + keyResult.id }}"
+        class="child"
```
Comment (peggimann) [inline frontend/src/app/shared/scoring/scoring/scoring.component.html:19]: maybe not use inline style

File: frontend/src/app/shared/scoring/scoring/scoring.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    it('should create', () => {
+      expect(component).toBeTruthy();
+    });
+  });
+});
+
+function createKeyResult(baseline: number, stretchGoal: number, lastCheckIn: CheckInMin | null) {
+  return {
+    id: 201,
+    title: 'Have more chocolate in office',
```
Comment (peggimann) [inline frontend/src/app/shared/scoring/scoring/scoring.component.spec.ts:131]: remove this method

File: frontend/src/app/shared/scoring/scoring/scoring.component.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  metricLabel: string = '';
+  isOverview: boolean = false;
+  labelWidth: string = '0';
+  endLineFail: string = '';
+  endLineCommit: string = '';
+  endLineTarget: string = '';
+  zoneWidth: number = 0;
+  checkInValue: number | string = 0;
+
+  keyResultCard: HTMLElement | null = null;
```
Comment (peggimann) [inline frontend/src/app/shared/scoring/scoring/scoring.component.ts:30]: maybe remove

File: frontend/src/style/styles.scss [hunk 5]
```diff
@@ -60,7 +60,7 @@ body {
 }
 
 .sub-title {
-  font-size: 0.7rem;
+  font-size: 8px;
```
Comment (peggimann) [inline frontend/src/style/styles.scss:63]: use rem instead

**Review by peggimann — CHANGES_REQUESTED** [review]

**Review by peggimann — APPROVED** [review]

**Review by peggimann — APPROVED** [review]

**Pull request merged**